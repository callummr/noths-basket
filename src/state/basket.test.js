import reducer, {
  incrementItem,
  decrementItem,
  deleteItem,
  setItemCount,
} from './basket'

describe('basket reducer', () => {
  it('should match default snapshot', () => {
    expect(reducer()).toMatchSnapshot()
  })

  it('should match incrementItem snapshot', () => {
    const incrementItemAction = incrementItem('004')

    expect(reducer({}, incrementItemAction)).toMatchSnapshot()
    expect(
      reducer({ '004': 3, '005': 1 }, incrementItemAction)
    ).toMatchSnapshot()
  })

  it('should match decrementItem snapshot', () => {
    const decrementItemAction = decrementItem('004')

    expect(reducer({}, decrementItemAction)).toMatchSnapshot()
    expect(
      reducer({ '004': 3, '005': 1 }, decrementItemAction)
    ).toMatchSnapshot()
  })

  it('should match deleteItem snapshot', () => {
    const deleteItemAction = deleteItem('004')

    expect(reducer({}, deleteItemAction)).toMatchSnapshot()
    expect(reducer({ '004': 3, '005': 1 }, deleteItemAction)).toMatchSnapshot()
  })

  it('should match setItemCount snapshot', () => {
    const setItemCountAction = setItemCount('004', 2)

    expect(reducer({}, setItemCountAction)).toMatchSnapshot()
    expect(
      reducer({ '004': 3, '005': 1 }, setItemCountAction)
    ).toMatchSnapshot()
  })
})

describe('products actions', () => {
  describe('incrementItem', () => {
    it('should match snapshot', () => {
      expect(incrementItem('004')).toMatchSnapshot()
    })
  })

  describe('decrementItem', () => {
    it('should match snapshot', () => {
      expect(decrementItem('004')).toMatchSnapshot()
    })
  })

  describe('deleteItem', () => {
    it('should match snapshot', () => {
      expect(deleteItem('004')).toMatchSnapshot()
    })
  })

  describe('setItemCount', () => {
    it('should match snapshot', () => {
      expect(setItemCount('004')).toMatchSnapshot()
    })
  })
})
