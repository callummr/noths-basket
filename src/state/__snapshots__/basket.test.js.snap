// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basket reducer should match decrementItem snapshot 1`] = `Object {}`;

exports[`basket reducer should match decrementItem snapshot 2`] = `
Object {
  "004": 2,
  "005": 1,
}
`;

exports[`basket reducer should match default snapshot 1`] = `Object {}`;

exports[`basket reducer should match deleteItem snapshot 1`] = `Object {}`;

exports[`basket reducer should match deleteItem snapshot 2`] = `
Object {
  "005": 1,
}
`;

exports[`basket reducer should match incrementItem snapshot 1`] = `
Object {
  "004": 1,
}
`;

exports[`basket reducer should match incrementItem snapshot 2`] = `
Object {
  "004": 4,
  "005": 1,
}
`;

exports[`basket reducer should match setItemCount snapshot 1`] = `
Object {
  "004": 2,
}
`;

exports[`basket reducer should match setItemCount snapshot 2`] = `
Object {
  "004": 2,
  "005": 1,
}
`;

exports[`products actions decrementItem should match snapshot 1`] = `
Object {
  "payload": "004",
  "type": "noths/basket/DECREMENT_ITEM",
}
`;

exports[`products actions deleteItem should match snapshot 1`] = `
Object {
  "payload": "004",
  "type": "noths/basket/DELETE_ITEM",
}
`;

exports[`products actions incrementItem should match snapshot 1`] = `
Object {
  "payload": "004",
  "type": "noths/basket/INCREMENT_ITEM",
}
`;

exports[`products actions setItemCount should match snapshot 1`] = `
Object {
  "payload": Object {
    "count": undefined,
    "id": "004",
  },
  "type": "noths/basket/SET_ITEM_COUNT",
}
`;
